src/nvim/eval/funcs.c

```diff
@@ -2751,16 +2751,16 @@ static void f_getpos(typval_T *argvars, typval_T *rettv, EvalFuncData fptr)
 /// Returns zero on error.
 static int getreg_get_regname(typval_T *argvars)
 {
-  const char_u *strregname;
+  const char *strregname;
 
   if (argvars[0].v_type != VAR_UNKNOWN) {
-    strregname = (const char_u *)tv_get_string_chk(&argvars[0]);
+    strregname = tv_get_string_chk(&argvars[0]);
     if (strregname == NULL) {  // type error; errmsg already given
       return 0;
     }
   } else {
     // Default to v:register
-    strregname = (char_u *)get_vim_var_str(VV_REG);
+    strregname = get_vim_var_str(VV_REG);
   }
 
   return *strregname == 0 ? '"' : *strregname;
```

Comment:
Doesn't matter currently, but returning a negative number doesn't seem to make sense here.
```suggestion
  return *strregname == 0 ? '"' : (uint8_t)(*strregname);
```